packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -732,6 +733,7 @@ async function handler(
         const users = await prisma.user.findMany({
           where: {
             username: { in: dynamicUserList },
+            organization: userOrgQuery(req.headers.host.replace(/^https?:\/\//, "")),
```

Comment:
When loading users we should only grab users that belong to the org. This is similar to how we fetch users on the dynamic booking page

Reply:
This doesn't feel like the correct place to do this code. i.e. Parsing request headers inside of place where we create a new booking to determine the org.

Do we not have this on session or somewhere? If not, feels like this should be pulled out to some other class and called here.

Reply:
I think this can only come from the request headers sadly.

Reply:
We don't have a session here (not guaranteed anyways) and we don't have an event type id. the only way really is parsed from the calling url. think that's the nicest way too

Reply:
I am not sure if I am right but I am reluctant towards sending either the `req`(more inclined towards it) or `req.headers` to userOrgQuery so that it can be sure that it is receiving **host** only and not some other string. We will get better type safety that way.

Reply:
Also, I feel that prisma client extensions would really solve this problem once and for all. We would be able to enforce an explicit value for `organization` there in user queries. Problem arises mostly when we forget to use `organization` in where.

Though not sure if I should focus on that considering, we are looking for prisma alternatives.

Reply:
I noticed that `req.headers.host` is being passed to `orgDomainConfig` which I think is similar to `userOrgQuery`.